cmd/argo/lint/lint.go

```diff
@@ -33,6 +34,9 @@ type LintOptions struct {
 	DefaultNamespace string
 	Formatter        Formatter
 	ServiceClients   ServiceClients
+	// Output if not nil, will print the linting results after
+	// source has been linted. Otherwise, nothing is printed.
+	Output io.Writer
```

Comment:
I think this probably be a parameter to `Summarzie`?

Reply:
Since we now call [RunLint](https://github.com/argoproj/argo-workflows/pull/5224/files#diff-e0caade70a58e8cd2f450d1f3e1baa498d2ac0e5713321d8169f0a747ee93f90R91) we no longer have to expose [Lint](https://github.com/argoproj/argo-workflows/pull/5224/files#diff-e0caade70a58e8cd2f450d1f3e1baa498d2ac0e5713321d8169f0a747ee93f90R115) and `LintOptions`. However, I think it would be useful to expose `Lint` from this package, because it could be utilized by 3rd party tools that use argo as a go module. WDYT?